### PR TITLE
24.07 perf

### PR DESCRIPTION
**Pull request by malay-nagda**
# What does this PR do ?

Documents performance benchmarks for LLM pre-training and finetuning using NeMo 24.07

**Collection**: [Run CICD]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [x] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: docs/source/performance_benchmarks.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+
+# Performance Benchmarks
+
+## Large Language Models**
+
+### Pretraining
+
+- The results in the table below show pre-training performance (using NeMo Framework 24.07) of various models on DGXH100, with FP8 precision.
```
Comment (erhoo82) [inline docs/source/performance_benchmarks.md:8]: I think it is better to create another bullet for the test environments used in the run.
For example, 
- Container: nemo 24.07 (link to the container)
- System: DGX-H100

File: docs/source/performance_benchmarks.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+- Please refer to `MLCommons Training results <https://mlcommons.org/benchmarks/training/>`_ for performance of GPT3-175B pre-training on large scale H100 systems.
```
Comment (erhoo82) [inline docs/source/performance_benchmarks.md:9]: Can we just add the performance MLPerf GPT3-1785B using the numbers and put the link directly on the number??

File: docs/source/performance_benchmarks.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+| Nemotron-8B   | 64     | 256  | 4   | 4096           | 2  | 1  | 1  | 12701              | 653                     | ***9***                                                |
+| Nemotron-22B  | 64     | 256  | 2   | 4096           | 2  | 4  | 1  | 4256               | 554                     | ***27***                                               |
+| Nemotron-340B | 128    | 32   | 1   | 4096           | 8  | 8  | 1  | 322                | 678                     | ***351***                                              |
+| LLAMA3-8B     | 8      | 128  | 1   | 8192           | 1  | 1  | 2  | 12036              | 697                     | ***9***                                                |
+| LLAMA3-70B    | 64     | 128  | 1   | 8192           | 4  | 4  | 2  | 1533               | 738                     | ***74***                                               |
+
+### Finetuning
+
+- The following table provides performance benchmarking of LLAMA2 models with SFT (supervised fine-tuning), and LoRA (Low-rank adaptors) on DGXH100, with FP8.
```
Comment (erhoo82) [inline docs/source/performance_benchmarks.md:26]: Let's have separate bullets for the system and container like the above.

File: docs/source/performance_benchmarks.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+- For fine-tuning, we use `SQuAD-v1.1 <https://rajpurkar.github.io/SQuAD-explorer/>`__ dataset, and the inputs are packed to 4096 tokens.
+
+
+| Model      | Mode     | #-GPUs | GBS | MBS | Sequence Length | TP | PP | Tokens / sec / GPU | Model TFLOP / sec / GPU |  ***Est. time to finetune in mins (10M tokens)***  |
```
Comment (erhoo82) [inline docs/source/performance_benchmarks.md:30]: Let's change `Sequence Length` to `Packed Sequence Length`

Comment (erhoo82) [reply docs/source/performance_benchmarks.md:30]: Let's change `Mode` to `Task`

**Comment by erhoo82** [comment]
Can we crate a new director like 'docs/source/performance/performance_summary.md`?

**Comment by erhoo82** [comment]
@malay-nagda Can we add a `Performance` section in the main README along with the link to this performance summary page? I think we can add more links in the main README if we want to highlight.

**Review by ericharper — APPROVED** [review]
